src/crimson/os/seastore/btree/fixed_kv_btree.h

```diff
@@ -511,6 +511,8 @@ class FixedKVBtree {
           &child_node);
       } else {
         if (i->get_val().pladdr.is_laddr()) {
+          assert(!node->children[i->get_offset()] ||
+                  is_reserved_ptr(node->children[i->get_offset()]));
```

Comment:
Thanks for the background!

A followup question, what's the difference between a nullptr and reserved-ptr child in case of an indirect mapping?

Or is it simpler not to distinguish between them (just use the nullptr) for the indirect mappings?

Reply:
> Thanks for the background!
> 
> A followup question, what's the difference between a nullptr and reserved-ptr child (of the leaf node) in case of an indirect mapping?
> 
> Or is it simpler not to distinguish between them (just use the nullptr) for the indirect mappings?

Please see the [comment ](https://github.com/ceph/ceph/pull/57828#discussion_r1660778601) above.